File: baekjoon/DP/2579.py

N = int(input())
numList = [0] * 301
for i in range(N):
    numList[i] = (int(input()))

dp = [0] * 301
dp[0] = numList[0]
dp[1] = numList[1] + numList[0]
dp[2] = max(numList[1]+numList[2], numList[0]+numList[2])

for i in range(3,N):
    dp[i] = max(dp[i-2] + numList[i], dp[i-3] + numList[i-1] + numList[i])

print(dp[N-1])

File: baekjoon_js/DP/2579.js
("use strict");

const fs = require("fs");
const stdin = (process.platform === "linux"
  ? fs.readFileSync("/dev/stdin").toString()
  : `6
10
20
15
25
10
20
`
).split("\n");

const input = (() => {
  let line = 0;
  return () => stdin[line++];
})();

let N = Number(input());

let stepList = Array(N + 1).fill(Number(0));
let dp = Array(N + 1).fill(Number(0));
for (let i = 0; i < N; i++) {
  let stepValue = Number(input());
  stepList[i + 1] = stepValue;
}

for (let i = 1; i < N + 1; i++) {
  //   console.log(dp);
  if (i === 1) {
    dp[i] = stepList[i];
  } else if (i === 2) {
    dp[i] = stepList[i] + stepList[i - 1];
  } else {
    dp[i] = Math.max(
      stepList[i] + stepList[i - 1] + dp[i - 3],
      dp[i - 2] + stepList[i]
    );
  }
}

// console.log(dp);
console.log(String(dp[N]));
